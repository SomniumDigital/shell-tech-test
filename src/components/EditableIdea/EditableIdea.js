import React  from 'react';
import deleteIcon from '../../images/trash-icon.svg';

const EditableIdea = ({
  edit,
  idea,
  editIdea,
  editedIdea,
  focusedIdea,
  updateInput,
  updateIdea,
  deleteIdea
}) => {
  const editHandler = (bool) => {
    editIdea(idea.id, bool)
  };

  const changeHandler = e => {
    updateInput(e.target.name, e.target.value, idea.id)
  };

  const blurHandler = () => {
    const title = editedIdea.ideaTitle || idea.title || 'Enter title';
    const body = editedIdea.ideaBody || idea.body || 'Enter idea';
    updateIdea(editedIdea.id, title, body)
  };

  const deleteHandler = () => {
    deleteIdea(idea.id)
  }

  return (
  <div>
    <div
      className="title"
      onMouseEnter={() => editHandler(true)}
      onMouseLeave={() => editHandler(false)}
    >
      { (editedIdea.id || focusedIdea) === idea.id ?
          <input
            type='text'
            name='ideaTitle'
            autoFocus
            placeholder={idea.title || 'Enter title'}
            value={editedIdea.ideaTitle || ''}
            onChange={changeHandler}
            onBlur={blurHandler}
          />
          : <h1>{idea.title || 'Enter title'}</h1> }
    </div>
    <div
      className="body"
      onMouseEnter={() => editHandler(true)}
      onMouseLeave={() => editHandler(false)}
    >
      { (editedIdea.id || focusedIdea) === idea.id ?
          <textarea
            name='ideaBody'
            placeholder={idea.body || 'Enter idea'}
            value={editedIdea.ideaBody || ''}
            onChange={changeHandler}
            onBlur={blurHandler}
          />
          : <p>{idea.body || 'Enter idea'}</p> }
    </div>
    <div className="delete">
      <button type="button" onClick={deleteHandler}>
        <img src={deleteIcon} alt="Delete" />
        Delete
      </button>
    </div>
  </div>
)};

export default EditableIdea;
