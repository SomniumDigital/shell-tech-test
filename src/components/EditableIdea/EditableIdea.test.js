import React  from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import EditableIdea from './EditableIdea';

configure({ adapter: new Adapter() });

describe('Component: EditableIdea', () => {
  it('renders correctly (with inputs)', () => {
    const component = shallow(
      <EditableIdea
        idea={{
          id: 'ABC123',
          title: 'Test',
          body: 'Test body'
        }}
        edit={false}
        editIdea={jest.fn()}
        editableIdeaId={'ABC123'}
        focusedIdea={null}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it('renders correctly (without inputs)', () => {
    const component = shallow(
      <EditableIdea
        idea={{
          id: 'ABC123',
          title: 'Test',
          body: 'Test body'
        }}
        edit={false}
        editIdea={jest.fn()}
        editableIdeaId={false}
        focusedIdea={null}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
