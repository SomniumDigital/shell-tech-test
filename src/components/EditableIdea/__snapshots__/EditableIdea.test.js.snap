// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: EditableIdea renders correctly (with inputs) 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditableIdea
    edit={false}
    editIdea={[Function]}
    editedIdea={
        Object {
            "id": false,
          }
    }
    focusedIdea={null}
    idea={
        Object {
            "body": "Test body",
            "id": "ABC123",
            "title": "Test",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
>
          <h1>
                    Test
          </h1>
</div>,
        <div
          className="body"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
>
          <p>
                    Test body
          </p>
</div>,
        <div
          className="delete"
>
          <button
                    onClick={[Function]}
                    type="button"
          >
                    <img
                              alt="Delete"
                              src="trash-icon.svg"
                    />
                    Delete
          </button>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Test
</h1>,
          "className": "title",
          "onMouseEnter": [Function],
          "onMouseLeave": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test",
          },
          "ref": null,
          "rendered": "Test",
          "type": "h1",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            Test body
</p>,
          "className": "body",
          "onMouseEnter": [Function],
          "onMouseLeave": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test body",
          },
          "ref": null,
          "rendered": "Test body",
          "type": "p",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            onClick={[Function]}
            type="button"
>
            <img
                        alt="Delete"
                        src="trash-icon.svg"
            />
            Delete
</button>,
          "className": "delete",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="Delete"
                src="trash-icon.svg"
/>,
              "Delete",
            ],
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Delete",
                "src": "trash-icon.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "Delete",
          ],
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
>
            <h1>
                        Test
            </h1>
</div>,
          <div
            className="body"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
>
            <p>
                        Test body
            </p>
</div>,
          <div
            className="delete"
>
            <button
                        onClick={[Function]}
                        type="button"
            >
                        <img
                                    alt="Delete"
                                    src="trash-icon.svg"
                        />
                        Delete
            </button>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Test
</h1>,
            "className": "title",
            "onMouseEnter": [Function],
            "onMouseLeave": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Test",
            },
            "ref": null,
            "rendered": "Test",
            "type": "h1",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              Test body
</p>,
            "className": "body",
            "onMouseEnter": [Function],
            "onMouseLeave": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Test body",
            },
            "ref": null,
            "rendered": "Test body",
            "type": "p",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              onClick={[Function]}
              type="button"
>
              <img
                            alt="Delete"
                            src="trash-icon.svg"
              />
              Delete
</button>,
            "className": "delete",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="Delete"
                  src="trash-icon.svg"
/>,
                "Delete",
              ],
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Delete",
                  "src": "trash-icon.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "Delete",
            ],
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Component: EditableIdea renders correctly (without inputs) 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditableIdea
    edit={false}
    editIdea={[Function]}
    editedIdea={
        Object {
            "id": false,
          }
    }
    focusedIdea={null}
    idea={
        Object {
            "body": "Test body",
            "id": "ABC123",
            "title": "Test",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
>
          <h1>
                    Test
          </h1>
</div>,
        <div
          className="body"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
>
          <p>
                    Test body
          </p>
</div>,
        <div
          className="delete"
>
          <button
                    onClick={[Function]}
                    type="button"
          >
                    <img
                              alt="Delete"
                              src="trash-icon.svg"
                    />
                    Delete
          </button>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Test
</h1>,
          "className": "title",
          "onMouseEnter": [Function],
          "onMouseLeave": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test",
          },
          "ref": null,
          "rendered": "Test",
          "type": "h1",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            Test body
</p>,
          "className": "body",
          "onMouseEnter": [Function],
          "onMouseLeave": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test body",
          },
          "ref": null,
          "rendered": "Test body",
          "type": "p",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            onClick={[Function]}
            type="button"
>
            <img
                        alt="Delete"
                        src="trash-icon.svg"
            />
            Delete
</button>,
          "className": "delete",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="Delete"
                src="trash-icon.svg"
/>,
              "Delete",
            ],
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Delete",
                "src": "trash-icon.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "Delete",
          ],
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
>
            <h1>
                        Test
            </h1>
</div>,
          <div
            className="body"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
>
            <p>
                        Test body
            </p>
</div>,
          <div
            className="delete"
>
            <button
                        onClick={[Function]}
                        type="button"
            >
                        <img
                                    alt="Delete"
                                    src="trash-icon.svg"
                        />
                        Delete
            </button>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Test
</h1>,
            "className": "title",
            "onMouseEnter": [Function],
            "onMouseLeave": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Test",
            },
            "ref": null,
            "rendered": "Test",
            "type": "h1",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              Test body
</p>,
            "className": "body",
            "onMouseEnter": [Function],
            "onMouseLeave": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Test body",
            },
            "ref": null,
            "rendered": "Test body",
            "type": "p",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              onClick={[Function]}
              type="button"
>
              <img
                            alt="Delete"
                            src="trash-icon.svg"
              />
              Delete
</button>,
            "className": "delete",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="Delete"
                  src="trash-icon.svg"
/>,
                "Delete",
              ],
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Delete",
                  "src": "trash-icon.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "Delete",
            ],
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
